src/compiler/irgen.jl

```diff
@@ -0,0 +1,455 @@
+    sig = Base.signature_type(ctx.f, ctx.tt)::Type
+    (ti, env) = ccall(:jl_type_intersection_with_env, Any,
+                      (Any, Any), sig, meth.sig)::Core.SimpleVector
+    meth = Base.func_for_method_checked(meth, ti)
+    linfo = ccall(:jl_specializations_get_linfo, Ref{Core.MethodInstance},
+                  (Any, Any, Any, UInt), meth, ti, env, world)
+
+    entry, modules = compile_linfo(ctx, linfo, world)
+
+    # link in dependent modules
+    mod = popfirst!(modules)
+    for dep in modules
+        link!(mod, dep)
+    end
+
+    # clean up incompatibilities
+    for llvmf in functions(mod)
+        llvmfn = LLVM.name(llvmf)
+
+        # only occurs in debug builds
+        delete!(function_attributes(llvmf), EnumAttribute("sspstrong", 0, JuliaContext()))
+
+        # dependent modules might have brought in other jfptr wrappers, delete them
+        if startswith(LLVM.name(llvmf), "jfptr_") && isempty(uses(llvmf))
+            unsafe_delete!(mod, llvmf)
+            continue
+        end
+
+        # llvmcall functions aren't to be called, so mark them internal (cleans up the IR)
+        if startswith(llvmfn, "jl_llvmcall")
+            linkage!(llvmf, LLVM.API.LLVMInternalLinkage)
+            continue
+        end
+
+        # rename functions
+        if !isdeclaration(llvmf)
+            # Julia disambiguates local functions by prefixing with `#\d#`.
+            # since we don't use a global function namespace, get rid of those tags.
+            if occursin(r"^julia_#\d+#", llvmfn)
+                llvmfn′ = replace(llvmfn, r"#\d+#"=>"")
+                if !haskey(functions(mod), llvmfn′)
+                    LLVM.name!(llvmf, llvmfn′)
+                    llvmfn = llvmfn′
+                end
+            end
+
+            # anonymous functions are just named `#\d`, make that somewhat more readable
+            m = match(r"_#(\d+)_", llvmfn)
+            if m !== nothing
+                llvmfn′ = replace(llvmfn, m.match=>"_anonymous$(m.captures[1])_")
+                LLVM.name!(llvmf, llvmfn′)
+                llvmfn = llvmfn′
+            end
+
+            # finally, make function names safe for ptxas
+            # (LLVM should to do this, but fails, see eg. D17738 and D19126)
+            llvmfn′ = safe_fn(llvmfn)
+            if llvmfn != llvmfn′
+                LLVM.name!(llvmf, llvmfn′)
+                llvmfn = llvmfn′
+            end
+        end
+    end
+
+    # rename the entry point
+    llvmfn = replace(LLVM.name(entry), r"_\d+$"=>"")
+    ## append a global unique counter
+    global globalUnique
+    globalUnique += 1
+    llvmfn *= "_$globalUnique"
+    LLVM.name!(entry, llvmfn)
+
+    # minimal required optimization
+    ModulePassManager() do pm
+        global global_ctx
+        global_ctx = ctx
+
+        add!(pm, ModulePass("LowerThrow", lower_throw!))
+        # TODO: Determine if these are useful, excise if not
+        #add!(pm, FunctionPass("HideUnreachable", hide_unreachable!))
```

Comment:
IIRC these work around limitations in ptxas

Reply:
They also make my `dotrap` debug function not work, so no objections from me.